dist/tools/compile_and_test_for_board/compile_and_test_for_board.py

```diff
@@ -303,11 +328,29 @@ def run_compilation_and_test(self, **test_kwargs):
         :returns: 0 on success and 1 on error.
         """
         try:
+            if self.testcase:
+                self.testcase.timestamp = time.time()
             self.compilation_and_test(**test_kwargs)
-            return None
+            res = None
         except ErrorInTest as err:
             self.logger.error('Failed during: %s', err)
-            return (str(err), err.application.appdir, err.errorfile)
+            res = (str(err), err.application.appdir, err.errorfile)
```

Comment:
I think we should distinguish between `pass failure error skipped`. It should be an error if flashing or building fails but if the test fails I think it should be a `failure`.

Reply:
Done. See 6853d3f7122de764c6deb34c89996ee314e56b94